src/main.cpp

```diff
@@ -4744,6 +4724,30 @@ PeerLogicValidation::PeerLogicValidation(CConnman* connmanIn) : connman(connmanI
     recentRejects.reset(new CRollingBloomFilter(120000, 0.000001));
 }
 
+void PeerLogicValidation::SyncTransaction(const CTransaction& tx, const CBlockIndex* pindex, int nPosInBlock) {
+    LOCK(cs_main);
```

Comment:
I agree, nice to move all logic to this function. 
But for now, we don't actually remove orphan txs which conflict with things accepted to mempool.  We could, however, move recursive processing of orphan txs that depended on mempool accepted txs here, I agree.
Also there are other places SyncTransaction is called, such as for disconnected block txs, where it would be incorrect to remove orphan txs which conflict with txs from disconnected blocks.
In any case, the code is correct now, I just want to be sure we think about all the ways SyncTransaction is called when we change it in the future.